File: src/main/java/com/itszuvalex/femtocraft/FemtocraftPlayerTracker.java
//package com.itszuvalex.femtocraft;
//
//import com.itszuvalex.femtocraft.sound.FemtocraftSoundManager;
//import cpw.mods.fml.common.IPlayerTracker;
//import cpw.mods.fml.common.network.PacketDispatcher;
//import cpw.mods.fml.common.network.Player;
//import net.minecraft.entity.player.EntityPlayer;
//
///**
// * Created by Chris on 9/3/2014.
// */
//public class FemtocraftPlayerTracker implements IPlayerTracker {
//    @Override
//    public void onPlayerLogin(EntityPlayer player) {
//
//    }
//
//    @Override
//    public void onPlayerLogout(EntityPlayer player) {
////        PacketDispatcher.sendPacketToPlayer(Femtocraft.soundManager.getPacket(STOP_ALL_SOUNDS), (Player) player);
//    }
//
//    @Override
//    public void onPlayerChangedDimension(EntityPlayer player) {
//        PacketDispatcher.sendPacketToPlayer(Femtocraft.soundManager().getPacket(FemtocraftSoundManager.SoundAction
//                .STOP_ALL_SOUNDS$.MODULE$), (Player) player);
//    }
//
//    @Override
//    public void onPlayerRespawn(EntityPlayer player) {
//
//    }
//}

File: src/main/scala/com/itszuvalex/femtocraft/FemtocraftPlayerTracker.scala
package com.itszuvalex.femtocraft

import cpw.mods.fml.common.eventhandler.SubscribeEvent
import cpw.mods.fml.common.gameevent.PlayerEvent.{PlayerChangedDimensionEvent, PlayerLoggedInEvent, PlayerLoggedOutEvent}

/**
 * Created by Chris on 9/3/2014.
 */
class FemtocraftPlayerTracker {

  @SubscribeEvent
  def onPlayerLogin(event: PlayerLoggedInEvent) = {
    Femtocraft.uuidManager.addMapping(event.player.getUniqueID, event.player.getCommandSenderName)
    Femtocraft.researchManager.onPlayerLogin(event.player)
  }

  @SubscribeEvent
  def onPlayerLogout(event: PlayerLoggedOutEvent) = {
    if (Femtocraft.soundManager != null) Femtocraft.soundManager.stopAllSounds()
  }

  @SubscribeEvent
  def onPlayerChangedDimension(event: PlayerChangedDimensionEvent) = if (Femtocraft.soundManager != null) Femtocraft.soundManager.stopAllSounds()


}

